src/lib.rs

```diff
@@ -1111,6 +1113,62 @@ impl TopicSpec {
             inner: NativeTopicSpec::new_computed(partitions, replications, ignore),
         }
     }
+
+    #[staticmethod]
+    pub fn new_mirror() -> TopicSpec {
+        let mut home_mirror = HomeMirrorConfig::from(vec![]);
+        //TODO: when update to 0.13.0
+        //home_mirror.source = home_to_remote;
+        let mirror_map = MirrorConfig::Home(home_mirror);
+        TopicSpec {
+            inner: NativeTopicSpec::new_mirror(mirror_map),
+        }
+    }
+
+    pub fn set_partition(&mut self, system: bool) {
+        self.inner.set_system(system);
+    }
+
+    pub fn set_system(&mut self, system: bool) {
+        self.inner.set_system(system);
+    }
+
+    pub fn set_retention_time(&mut self, time_in_seconds: i64) {
+        self.inner
+            .set_cleanup_policy(CleanupPolicy::Segment(SegmentBasedPolicy {
+                time_in_seconds: time_in_seconds as u32,
+            }));
+    }
+
+    pub fn set_storage(&mut self, max_partition_size: Option<i64>, segment_size: Option<i64>) {
+        let mut storage = TopicStorageConfig::default();
+
+        if let Some(segment_size) = segment_size {
+            storage.segment_size = Some(segment_size as u32);
+        }
+
+        if let Some(max_partition_size) = max_partition_size {
+            storage.max_partition_size = Some(max_partition_size as u64);
+        }
+
+        self.inner.set_storage(storage);
+    }
+
+    pub fn set_compression_type(&mut self, compression: String) {
+        if compression == "none" {
+            self.inner.set_compression_type(CompressionAlgorithm::None);
+        } else if compression == "gzip" {
+            self.inner.set_compression_type(CompressionAlgorithm::Gzip);
+        } else if compression == "snappy" {
+            self.inner
+                .set_compression_type(CompressionAlgorithm::Snappy);
+        } else if compression == "lz4" {
+            self.inner.set_compression_type(CompressionAlgorithm::Lz4);
+        } else if compression == "zstd" {
+            self.inner.set_compression_type(CompressionAlgorithm::Zstd);
+        }
+        self.inner.set_compression_type(CompressionAlgorithm::Any);
+    }
```

Comment:
This should probably return an error if not recognized, or it could accept a compression type enum

Reply:
I changed to return result, and an error for this case.
The support for enums that are not tag unions does not seem great in pyo3, it's also not automatically typed, so no benefits at all.